### PR TITLE
Implementation of Pareto Distribution

### LINKED ISSUE #4434: [Feature Request] Implement Pareto Distribution
I could issue a PR if you feel this is worth adding to the list of distributions.

### PR DESCRIPTION
**Pull request by vishwakftw**
This closes https://github.com/pytorch/pytorch/issues/4434 .

Thanks to @fritzo. Math checked by @fritzo

### CODE REVIEW

**Comment by vishwakftw** [comment]
`_check_sampler_sampler` seems to be causing problems in all of the builds except `pynightly`. @fritzo , any suggestions?

**Comment by fritzo** [comment]
Have you run tests locally?

**Comment by fritzo** [comment]
@vishwakftw Before submitting a PR you should run distributions tests locally:
```
python tests/test_distributions.py
```
or if you prefer pytest
```py
pytest -v tests/tests_distributions.py
```

**Comment by fritzo** [comment]
@vishwakftw The test that is failing is a two-sample quantile test and indicates that there is probably a bug in either your `.rsample()` math or in your usage of `scipy.stats.pareto`. I recommend visualizing quantile-quantile plots to help diagnose, e.g. [here](https://github.com/fritzo/notebooks/blob/master/debug-gamma-sampler.ipynb) is a notebook I've used to debug the Gamma sampler.

**Comment by vishwakftw** [comment]
@fritzo I am extremely sorry about the inconvenience caused. There was a minor bug in the `rsample` method (I forgot to divide by `self.alpha`), which led to these issues.

**Comment by fritzo** [comment]
@vishwakftw No problem, I should have made it clearer how to test locally and confirmed that you could reproduce the bug. Glad things are working now.

**Review by fritzo — APPROVED** [review]

File: test/test_distributions.py [hunk 1]
```diff
@@ -101,6 +87,30 @@
...
+        {
+            'alpha': Variable(torch.exp(torch.randn(2, 3)), requires_grad=True),
+            'beta': Variable(torch.exp(torch.randn(2, 3)), requires_grad=True),
+        },
+        {
+            'alpha': Variable(torch.exp(torch.randn(1)), requires_grad=True),
+            'beta': Variable(torch.exp(torch.randn(1)), requires_grad=True),
+        },
+    ]),
+    Example(Laplace, [
```
Comment (fritzo) [inline test/test_distributions.py]: FYI while it's nice to keep these in alphabetical order, by moving these around you will cause merge conflicts for all the other PRs that are currently under review.

**Comment by apaszke** [comment]
Thanks @vishwakftw!

**Pull request merged**